util/admin/admin.go

```diff
@@ -181,6 +181,11 @@ func CancelJobs(txn kv.Transaction, ids []int64) ([]error, error) {
 			} else {
 				err = t.UpdateDDLJob(int64(j), job, true)
 			}
+			if err != nil {
+				errs[i] = errors.Trace(err)
+				continue
+			}
+			err = t.AddCancellingJobList(job)
```

Comment:
better check the duplicate there, if already in, just continue

Reply:
If we check the duplicate here, it seems that the second `removeJobFromCancellingList` mentioned above can be removed.